packages/eas-cli/src/branch/queries.ts

```diff
@@ -0,0 +1,114 @@
+import chalk from 'chalk';
+import CliTable from 'cli-table3';
+
+import { PaginatedQueryFlags } from '../commandUtils/flagHelpers';
+import { UpdateBranchFragment } from '../graphql/generated';
+import { BranchQuery } from '../graphql/queries/BranchQuery';
+import Log from '../log';
+import { UPDATE_COLUMNS, formatUpdate, getPlatformsForGroup } from '../update/utils';
+import { printJsonOnlyOutput } from '../utils/json';
+import {
+  PaginatedQueryPromptOptions,
+  PaginatedQueryPromptType,
+  PaginatedQueryResponse,
+  performPaginatedQueryAsync,
+} from '../utils/queries';
+
+export const BRANCHES_LIMIT = 50;
+
+export async function selectBranchForProjectAsync(
+  projectId: string,
+  promptTitle: string,
+  flags: PaginatedQueryFlags
+): Promise<UpdateBranchFragment> {
+  if (flags['non-interactive']) {
+    throw new Error('Unable to select a branch in non-interactive mode.');
+  }
+
+  const fromatUpdateBranchFragment = (updateBranchFragment: UpdateBranchFragment): string =>
+    updateBranchFragment.name;
+  const getIdentifierForQueryItem = (updateBranchFragment: UpdateBranchFragment): string =>
+    updateBranchFragment.id;
+
+  const branch = await performPaginatedQueryAsync({
+    pageSize: flags.limit ?? BRANCHES_LIMIT,
+    offset: flags.offset ?? 0,
+    queryToPerform: queryBranchesForProject(projectId),
+    promptOptions: {
+      type: PaginatedQueryPromptType.select,
+      title: promptTitle,
+      getIdentifierForQueryItem,
+      createDisplayTextForSelectionPromptListItem: fromatUpdateBranchFragment,
+    },
+  });
+  return branch.pop()!;
+}
+
+export async function queryForBranchesAsync(
+  projectId: string,
+  flags: PaginatedQueryFlags
+): Promise<void> {
+  const promptOptions: PaginatedQueryPromptOptions<UpdateBranchFragment> =
+    flags['non-interactive'] || flags.limit
+      ? { type: PaginatedQueryPromptType.none, renderQueryResults: renderPageOfBranches(flags) }
+      : {
+          type: PaginatedQueryPromptType.confirm,
+          title: 'Load more branches?',
+          renderListItems: renderPageOfBranches(flags),
+        };
+
+  await performPaginatedQueryAsync({
+    pageSize: flags.limit ?? BRANCHES_LIMIT,
+    offset: flags.offset ?? 0,
```

Comment:
i also dont feel strongly here. i did not do this originally because i wanted to prevent consumers from having to import `branches_limit` in addition to the `queryForBranchesAsync` function (thus making call sites more verbose and providing potential to go out of sync with other call sites). the mentality is "i want to do as much shared 'set-up' as possible for the end consumer". 

> Or at least, this should be extracted to a common place, so both selectBranchForProjectAsync and queryForBranchesAsync use exactly the same logic. 

I do think this makes sense. I just didn't see them getting out of sync since they were co-located in the same file and have been properly configured in this PR.